src/main/java/hudson/plugins/gradle/injection/SetMavenOpts.java

```diff
@@ -0,0 +1,30 @@
+package hudson.plugins.gradle.injection;
+
+import hudson.EnvVars;
+import jenkins.security.MasterToSlaveCallable;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+public class SetMavenOpts extends MasterToSlaveCallable<Void, RuntimeException> {
```

Comment:
We should try whether we can set the environment from `MavenBuildScanInjection` by calling 
```java
node.getNodeProperties().add(new EnvironmentVariablesNodeProperty(new EnvironmentVariablesNodeProperty.Entry("MAVEN_OPTS", "the value")));
```